File: Arcade/Intro/C++/08_matrixElementsSum.cpp
int matrixElementsSum(std::vector<std::vector<int>> matrix) {
    int sum=0;
        for(int i=0;i<matrix[0].size();i++){
                for(int j=0;j<matrix.size();j++){
                        if(matrix[j][i]==0) break;
                        sum+=matrix[j][i];
                }
        }
        return sum;
}

File: Arcade/Intro/JavaScript/08_matrixElementsSum.js
function matrixElementsSum(matrix) {
    var total = matrix[0].reduce(sum, 0);
    
    for (var i = 1; i < matrix.length; i++) {
        for (var j = 0; j < matrix[0].length; j++) {
            if (matrix[i-1][j] === 0) {
                matrix[i][j] = 0;
            } else {
                total += matrix[i][j];
            }
        }
    }
    return total;
    
} function sum(a, b) {
    return a + b;
}
